httpserver/http_connection.cpp

```diff
@@ -64,32 +64,40 @@ void make_http_request(boost::asio::io_context& ioc, const std::string& address,
                        uint16_t port, const std::shared_ptr<request_t>& req,
                        http_callback_t&& cb) {
 
-    error_code ec;
-    tcp::endpoint endpoint;
-    tcp::resolver resolver(ioc);
+    static tcp::resolver resolver(ioc);
```

Comment:
In general please avoid non-constant global variables; see https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ri-global for discussion.

In this particular case `resolver`'s lifetime outlives `ioc`'s, which is unsafe (and will probably segfault if there are any outstanding resolve requests when shutdown occurs).  But also `tcp::resolver` construction does nothing (it just stores a reference) so this isn't optimizing anything.

Reply:
I put it in `shared_ptr`, which should extend its lifetime until it is no longer needed (I think).